src/view/input_parsers/request_data.rs

```diff
@@ -205,6 +265,167 @@ mod parsers_request_items {
 pub mod tests_parsers_request_items {
     use super::*;
 
+    #[test]
+    fn test_string_json_value() {
+        let input = "married:=\"abc\"";
+        let base_request = PartialRequestData::default().with_body("".to_string());
+
+        assert_eq!(
+            None,
+            parsers_request_items::non_string_body_value(input, &base_request)
+        )
+    }
+
+    #[test]
+    fn test_number_json_value() {
+        let input = "age:=29";
+        let base_request = PartialRequestData::default().with_body("".to_string());
+
+        let expected_result =
+            PartialRequestData::default().with_body(r#"{ "age": 29 }"#.to_string());
+
+        assert_eq!(
+            Some(expected_result),
+            parsers_request_items::non_string_body_value(input, &base_request)
+        );
+    }
+
+    #[test]
+    fn test_pos_number_json_value() {
+        let input = "age:=+29";
+        let base_request = PartialRequestData::default().with_body("".to_string());
+
+        assert_eq!(
+            None,
+            parsers_request_items::non_string_body_value(input, &base_request)
+        );
+    }
+
+    #[test]
+    fn test_neg_number_json_value() {
+        let input = "age:=-29";
+        let base_request = PartialRequestData::default().with_body("".to_string());
+
+        let expected_result =
+            PartialRequestData::default().with_body(r#"{ "age": -29 }"#.to_string());
+
+        assert_eq!(
+            Some(expected_result),
+            parsers_request_items::non_string_body_value(input, &base_request)
+        );
+    }
+
+    #[test]
+    fn test_float_number_json_value() {
+        let input = "age:=29.0";
+        let base_request = PartialRequestData::default().with_body("".to_string());
+
+        let expected_result =
+            PartialRequestData::default().with_body(r#"{ "age": 29.0 }"#.to_string());
+
+        assert_eq!(
+            Some(expected_result),
+            parsers_request_items::non_string_body_value(input, &base_request)
+        );
+    }
+
+    #[test]
+    fn test_pos_float_number_json_value() {
+        let input = "age:=+29.5";
+        let base_request = PartialRequestData::default().with_body("".to_string());
+
+        assert_eq!(
+            None,
+            parsers_request_items::non_string_body_value(input, &base_request)
+        );
+    }
+
+    #[test]
+    fn test_neg_float_number_json_value() {
+        let input = "age:=-29.5";
+        let base_request = PartialRequestData::default().with_body("".to_string());
+
+        let expected_result =
+            PartialRequestData::default().with_body(r#"{ "age": -29.5 }"#.to_string());
+
+        assert_eq!(
+            Some(expected_result),
+            parsers_request_items::non_string_body_value(input, &base_request)
+        );
+    }
```

Comment:
Merge all these tests in the same function. Once they all are number validaton with tiny variation. You can follow the template of tests like this: 
- https://github.com/talis-fb/TReq/blob/6814b061f2cc010422e8ca42bf4697d7baf751fa/src/view/input_parsers/request_data.rs#L269
- https://github.com/talis-fb/TReq/blob/6814b061f2cc010422e8ca42bf4697d7baf751fa/src/view/input/validators/url_alias.rs#L58

It would be nice create tests function for each "possible value" and create the "cases" inside function. Make a test function for each case make them scattered (harder to find a specific one) and the overall file verbose.